packages/splitText/src/splitIntoWords.mjs

```diff
@@ -1,2 +1,27 @@
-// Hypothesis: A word consists of non-space characters, followed by any number of space characters.
-export default (text) => text.match(/\s*\S+\s*/g) || [text];
+// Break words at spaces and dashes.
+// Why a dashes? If we don't, long words connected by dashes (especially in German) will blow
+// the layout as lines become longer than the layout might be wide.
+export default (text) => {
+    // This splits text into words and whatever's inbetween; 'hello-world there' becomes
+    // ['hello', '-', 'world', ' ', 'there']
+    // Parens are necessary to include the boundaries in the result
+    const boundaryRegex = /([\s-]+)/g;
+    // Remove all empty parts that might be added by using split
+    const textSplitAtBoundaries = text.split(boundaryRegex).filter((part) => part !== '');
+    const words = [];
+    textSplitAtBoundaries.forEach((part, index) => {
+        const isBoundary = boundaryRegex.test(part);
+        if (index === 0) words.push([part]);
+        // If the string starts with a boundary, exceptionally add the word after it to the
+        // previous boundary; if we don't, the first word would consist of the boundary only.
+        else if (index === 1 && !isBoundary) words.at(-1).push(part);
```

Comment:
🤔 Verstehe ich das richtig, du rufst hier `.push()` auf einem String auf?

Reply:
Nicht ganz: Auf L14 und L21 pushe ich jeweils ein Array in das Array, keinen String. Ich pushe deswegen hier auch in das Child-Array von `words`.